### PR TITLE
F t31336 adjust and add contracts

### PR DESCRIPTION
**Pull request by bakridemos**
Ticket: T31336

Description: add needed contracts

### CODE REVIEW

File: src/Contracts/Entities/ElementsInterface.php [hunk 1]
```diff
@@ -6,5 +6,67 @@
 
 interface ElementsInterface
 {
+    public const ELEMENTS_CATEGORY_MAP = 'map'; // like "Planzeichnung"
+    public const ELEMENTS_CATEGORY_STATEMENT = 'statement'; // like "Gesamtstellungnahme" or "Fehlanzeige"
+    public const ELEMENTS_CATEGORY_FILE = 'file'; // like "Ergänzende Unterlagen" or "Landschaftsplan-Änderung"
+    public const ELEMENTS_CATEGORY_PARAGRAPH = 'paragraph'; // like "Begründung" or "Textliche Festsetzungen"
+    public const ELEMENTS_CATEGORY_CATEGORY = 'category'; // created element by customer
 
+    public const FILE_TYPE_ANSCHREIBEN = 'Anschreiben';
```
Comment (mussbach) [inline src/Contracts/Entities/ElementsInterface.php:15]: shouldn't this rather be in the specific addon?

File: src/Contracts/Exceptions/AddonContentMandatoryFieldsException.php [hunk 2]
```diff
@@ -0,0 +1,25 @@
+<?php
+
+namespace DemosEurope\DemosplanAddon\Contracts\Exceptions;
+
+use Exception;
```
Comment (mussbach) [inline src/Contracts/Exceptions/AddonContentMandatoryFieldsException.php:5]: `use Throwable;` missing

**Review by mussbach — CHANGES_REQUESTED** [review]

File: src/Contracts/Entities/ElementsInterface.php (hunk 1 above)
Comment (bakridemos) [reply src/Contracts/Entities/ElementsInterface.php:15]: they are used in core too, they have to be hier to be availble for both, Xbauleitplanung addon and the core

File: src/Contracts/Exceptions/AddonContentMandatoryFieldsException.php (hunk 2 above)
Comment (bakridemos) [reply src/Contracts/Exceptions/AddonContentMandatoryFieldsException.php:5]: thx, done in https://github.com/demos-europe/demosplan-addon/pull/32/commits/9893a5074d2666845964ab2a5e044b8321766a8e

File: src/Contracts/Entities/ElementsInterface.php (hunk 1 above)
Comment (mussbach) [reply src/Contracts/Entities/ElementsInterface.php:15]: But the ones below are not

Comment (bakridemos) [reply src/Contracts/Entities/ElementsInterface.php:15]: u are right, there are many constants that are not used ( not in core and not in addon), i thought it is better to keep them all together, so i moved them all in ElementsInterface.

Comment (mussbach) [reply src/Contracts/Entities/ElementsInterface.php:15]: The ones that are not used in core are all potentially used in XBauleiplanung addon. I would move all of them there.

Comment (bakridemos) [reply src/Contracts/Entities/ElementsInterface.php]: done in https://github.com/demos-europe/demosplan-addon/pull/32/commits/fdc250fd52d2001b2519140b8c401eeba3c2d3e1

**Review by graupnerdemos — APPROVED** [review]
99% lgtm, 1% accepted with marked hesitation.

**Review by mussbach — APPROVED** [review]

**Pull request merged**